src/goto-symex/call_stack.h

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_GOTO_SYMEX_CALL_STACK_H
+#define CPROVER_GOTO_SYMEX_CALL_STACK_H
+
+#include "frame.h"
+
+class call_stackt : public std::vector<framet>
+{
+public:
+  framet &top()
+  {
+    PRECONDITION(!empty());
+    return back();
+  }
+
+  const framet &top() const
+  {
+    PRECONDITION(!empty());
+    return back();
+  }
+
+  framet &new_frame(symex_targett::sourcet calling_location)
+  {
+    emplace_back(calling_location);
+    return back();
+  }
+
+  void pop()
+  {
+    pop_back();
```

Comment:
A `PRECONDITION(!empty())` here would be appreciated, as there's no guarantee that this can not be called on an empty call stack.